crates/solvers/src/boundary/liquidity/limit_order.rs

```diff
@@ -0,0 +1,155 @@
+use {
+    crate::domain::liquidity::limit_order::LimitOrder,
+    contracts::ethcontract::{H160, U256},
+    shared::{baseline_solver::BaselineSolvable, price_estimation::gas::GAS_PER_ZEROEX_ORDER},
+};
+
+// Follows 0x's contract implementation: <https://github.com/0xProject/protocol/blob/%400x/contracts-utils%404.8.52/contracts/utils/contracts/src/v06/LibMathV06.sol#L71-L83>
+impl BaselineSolvable for LimitOrder {
+    fn get_amount_out(&self, out_token: H160, (in_amount, in_token): (U256, H160)) -> Option<U256> {
+        if in_token != self.taker.token.0
+            || out_token != self.maker.token.0
+            || in_amount > self.taker.amount
+        {
+            return None;
+        }
+
+        in_amount
```

Comment:
Remind me again, are all 0x liquidity orders partially fillable? Otherwise we probably need to check that here.

Reply:
Not sure it is stated in the [docs](https://docs.0xprotocol.org/en/latest/basics/orders.html?highlight=how%20to%20sign#limit-orders) explicitly.
Found [this](https://github.com/0xProject/0x-protocol-specification/blob/master/v3/v3-specification.md#fillorder).